slam-libraries/viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -101,18 +101,25 @@ ::grpc::Status SLAMServiceImpl::GetPointCloudMap(
     try {
         std::shared_lock optimization_lock{optimization_shared_mutex,
                                            std::defer_lock};
-        if (optimization_lock.try_lock()) {
+        if ((optimization_lock.try_lock()) && (!localization_map_ready)) {
```

Comment:
You should check `localization_map_ready` before trying to acquire the lock (i.e. put it first in the if-statement).